sdk/identity/identity/src/msal/utils.ts

```diff
@@ -74,8 +74,12 @@ export function getAuthority(tenantId: string, host?: string): string {
  * by sending it within the known authorities in the MSAL configuration.
  * @internal
  */
-export function getKnownAuthorities(tenantId: string, authorityHost: string): string[] {
-  if (tenantId === "adfs" && authorityHost) {
+export function getKnownAuthorities(
+  tenantId: string,
+  authorityHost: string,
+  instanceDiscovery?: boolean
+): string[] {
+  if ((tenantId === "adfs" && authorityHost) || instanceDiscovery) {
```

Comment:
If "instance discovery" is enabled (which is what happens today), you let MSAL handle this. If instance discoevry is disabled, you declare that the input authority is "known"

```suggestion
  if ((tenantId === "adfs" && authorityHost) || !instanceDiscovery) {
```

Reply:
Not sure when `authorityHost` will be null though. There is always an authority. If you don't specify one, MSAL will use "https://login.microsoftonline.com/common"